src/helix-ui/elements/_hx-checkbox.less

```diff
@@ -19,12 +44,91 @@
   width: 1em;
 }
 
-:host {
-  &([checked]:not([indeterminate])) #tick {
-    display: block;
+#nativeControl:checked:not(:indeterminate) ~ #customControl #tick {
+  display: block;
+}
+
+#nativeControl:indeterminate ~ #customControl #minus {
+  display: block;
+}
+
+#nativeControl {
+  /* opacity 0 because Firefox and OS focus styles */
+  opacity: 0;
+  z-index: 0;
+
+  &:focus {
+    outline: none;
+
+    ~ #customControl {
+      border-color: @cyan-700;
+      box-shadow: @focus-glow;
+    }
   }
 
-  &([indeterminate]) #minus {
-    display: block;
+  /* default checked */
+  &:checked ~ #customControl,
+  /* default indeterminate (checked or unchecked) */
+  &:indeterminate ~ #customControl {
+    color: @cyan-900;
+
+    &:hover {
+      background-color: @cyan-50;
+      color: @cyan-500;
+    }
   }
-}
+
+  /* disabled unchecked */
+  &:disabled ~ #customControl {
+    background-color: @gray-100;
+    color: @gray-500;
+    cursor: not-allowed;
+
+    & + label {
+      &:extend(.hxSubdued);
+    }
+
+    &:hover {
+      background-color: @gray-100;
+      color: @gray-500;
+    }
+  }//[disabled]
+}//input
+
+/* invalid */
+:host([invalid]){
+  #customControl {
+    border-width: 2px;
+    color: @red-900;
+
+    &:hover {
+      background-color: @red-100;
+    }
+  }
+
+  #nativeControl:focus ~ #customControl {
+    border-color: @red-900;
+    box-shadow: @focus-glow-invalid;
+  }
+
+  /* invalid and checked or indeterminate */
+  #nativeControl:checked ~ #customControl, 
+  #nativeControl:indeterminate ~ #customControl {
+    border-width: 2px;
+    color: @red-900;
```

Comment:
Double-check. If override is needed, please leave a comment in the code explaining that it's needed.